### ISSUE TITLE
Change `f^-1` to only work for bundled equivalences

### ISSUE THREAD

**Issue by Alizter** [HoTT/HoTT#1240]
As discussed in #1165, #1238 and elsewhere, it would be beneficial for the notation `^-1` to only mean `equiv_inverse`. Otherwise currently coq spends a lot of time searching for IsEquiv instances when it really should be carrying them around. There are a few places in the library, notably my cubical library which I have made note of #1217, that use unbundled equivalences and invert them. These will have to be bundled up then inverted. We should expect to see a performance increase.

**Comment by Alizter** [comment]
In #1252 I ran into issues where coq was typechecking `f^-1`. If I checked the entire proof it would hang, but if I stepped through it would succeed. `f^-1` is in hindsight a terrible notion because it causes coq to unfold `f` when looking for instances of `IsEquiv f`. This can get messy, especially if `f` has some troublesome insides. Whats even worse is that in many cases it is `^-1%equiv` that was really intended.

**Comment by mikeshulman** [comment]
i started trying to do this, but there is a problem: `cbn` unfolds `f^-1 x` to `equiv_inv f x`, which we don't want to have to look at.  Is there a way to prevent that?

**Comment by Alizter** [comment]
Well my plan was to have two notations like we currently do, but somehow make the `equiv` one the "default" one. I wonder if we could make an "isequiv_scope" or something. Then we would open equiv_scope on top of isequiv_scope.

**Comment by mikeshulman** [comment]
Maybe this is a job for someone who understands scopes better than I do.

**Comment by Alizter** [comment]
This doesn't seem like an easy thing to do after having tried it a few times. I will mess around with it tomorrow.

**Comment by mikeshulman** [comment]
I wonder if @fpvandoorn has any ideas.

**Comment by mikeshulman** [comment]
This isn't the first time I've been annoyed that Coq doesn't have a way to set a notation as "only displaying".

**Comment by mikeshulman** [comment]
I guess one possibility -- probably not a good one -- is to define `Equiv` as a big record with all the fields in it directly, and then define `IsEquiv f` to mean that there is an `Equiv` whose underlying function is equal to `f`.

**Comment by SkySkimmer** [comment]
>This isn't the first time I've been annoyed that Coq doesn't have a way to set a notation as "only displaying".

Isn't that what `only printing` is for? (the note just before https://coq.github.io/doc/master/refman/user-extensions/syntax-extensions.html#the-infix-command)

**Comment by mikeshulman** [comment]
Huh!  It does look that way.  That must be relatively new; I could swear it didn't used to exist.

**Comment by fpvandoorn** [comment]
I think you can tell `cbn` to never unfold `f^-1 % equiv`: https://coq.inria.fr/refman/proof-engine/tactics.html#coq:tacn.cbn
But maybe that is undesirable if `f` is a defined equivalence with a simple inverse.

You could define a second notation `f^-1 x` for `equiv_inv f x`. Was that the intended application for the "only printing" notation? Though it is potentially a little dangerous to have one notation for two different things (even if they are definitionally equal): maybe rewrite fails because one is written and the other is expected.

**Comment by mikeshulman** [comment]
Yes, I think we do want it to get unfolded.  And yes, that's what I was thinking of with the only printing notation.  That's a good point about rewrite... but would it be any worse than the current situation, where we already have the one notation meaning both things?

**Comment by mikeshulman** [comment]
Okay, well, I tried this a bit, but it looks like would require a lot of redesign work and thought that I don't really have time for, and the current setup isn't broken enough for me to be really motivated to change it.  But anyone else who wants to give it a go is welcome to.

**Comment by mikeshulman** [comment]
Perhaps a simpler partial solution would be to just have _different_ notations: `^-1` for bundled equivalences and, say, `^-1'` for unbundled ones.  That might act mostly in practice like our current `^-1%equiv` and `^-1`, but with the nicer notation for the better-behaved one, encouraging people to use it preferentially.
